server/clients.go

```diff
@@ -142,7 +142,7 @@ func (s *IDPServer) serveClients(w http.ResponseWriter, r *http.Request) {
 	s.mu.Unlock()
 
 	if !ok {
-		http.Error(w, "tsidp: client not found", http.StatusNotFound)
+		writeHTTPError(w, r, http.StatusNotFound, "not_found", "tsidp: client not found")
```

Comment:
nit: to match a similar error message from `authorize.go`
```suggestion
		writeHTTPError(w, r, http.StatusNotFound, "not_found", "tsidp: invalid client ID")
```